### PR TITLE
Add TechnicPad macropad

### PR DESCRIPTION
**Pull request by nwhirschfeld**
## Description

Added [TechnicPad](https://github.com/nwhirschfeld/TechnicPad) keyboard (macropad) and the keymaps _default_ and _via_.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* none

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/technicpad/rules.mk [hunk 1]
```diff
@@ -0,0 +1,4 @@
+BOOTMAGIC_ENABLE 	= yes   # Enable Bootmagic Lite
+MOUSEKEY_ENABLE 	= yes   # Mouse keys
+EXTRAKEY_ENABLE 	= yes	# Audio control and System control
+LTO_ENABLE 		= yes
```
Comment (zvecr) [inline keyboards/technicpad/rules.mk:4]:
```suggestion
# This file intentionally left blank
```

File: keyboards/technicpad/readme.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# TechnicPad
+
+![TechnicPad](https://i.imgur.com/LPD0KSL.jpg)
+
+A 4% (2x2) macropad, that is compatible to technic building blocks. 
+
+* Keyboard Maintainer: [Niclas Hirschfeld](https://github.com/nwhirschfeld)
+* Hardware Supported: Pro Micro boards
+* Hardware Availability: for now TechnicPad is handwired and uses an Arduino pro micro. [TechnicPad repository](https://github.com/nwhirschfeld/TechnicPad)
```
Comment (zvecr) [inline keyboards/technicpad/readme.md:9]: Should be relocated to `keyboards/handwired`

File: keyboards/technicpad/keymaps/default/readme.md
Comment (zvecr) [inline keyboards/technicpad/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/technicpad/keymaps/via/rules.mk
Comment (zvecr) [inline keyboards/technicpad/keymaps/via/rules.mk:1]: This file should be removed.

File: keyboards/technicpad/keymaps/via/keymap.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "keyboard": "technicpad",
+    "keymap": "default",
```
Comment (zvecr) [inline keyboards/technicpad/keymaps/via/keymap.json:3]:
```suggestion
    "keymap": "via",
    "config": {
        "features": {
            "via": true
        }
    },
```

File: keyboards/technicpad/info.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+    "manufacturer": "nwhirschfeld",
+    "keyboard_name": "TechnicPad",
+    "maintainer": "nwhirschfeld",
+    "bootloader": "caterina",
+    "bootmagic": {
+        "matrix": [3, 0]
```
Comment (zvecr) [inline keyboards/technicpad/info.json:7]: This does not match the comment within the readme.

> * **Bootmagic reset**: Hold down the upper left key and plug in the keyboard

File: keyboards/technicpad/readme.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    make technicpad:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the upper left key and plug in the keyboard
+* **Physical reset**: Short the _RST_ and _GND_ pins of the PCB
```
Comment (zvecr) [inline keyboards/technicpad/readme.md:26]:
```suggestion
* **Physical reset**: Short the _RST_ and _GND_ pins of the PCB
* **Keycode in layout**: Press the key mapped to QK_BOOT if it is available
```

File: keyboards/technicpad/info.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    },
```
Comment (zvecr) [inline keyboards/technicpad/info.json:8]:
```suggestion
    },
    "build": {
        "lto": true
    },
    "features": {
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true
    },
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/technicpad/keymaps/default/readme.md
Comment (nwhirschfeld) [reply keyboards/technicpad/keymaps/default/readme.md:1]: done in https://github.com/nwhirschfeld/qmk_firmware/commit/f4a9bf345b379f46545247a34f18112459ba1f1d

File: keyboards/technicpad/info.json (hunk 4 above)
Comment (nwhirschfeld) [reply keyboards/technicpad/info.json]: resolved in https://github.com/nwhirschfeld/qmk_firmware/commit/7cba09e585de78caee4bd05af1534f4a54ef25a3

File: keyboards/technicpad/readme.md (hunk 2 above)
Comment (nwhirschfeld) [reply keyboards/technicpad/readme.md]: https://github.com/nwhirschfeld/qmk_firmware/commit/b12a2cd4e00fd9e266def8c60ab2cae7dfaf2893

File: keyboards/technicpad/keymaps/via/rules.mk
Comment (nwhirschfeld) [reply keyboards/technicpad/keymaps/via/rules.mk:1]: https://github.com/nwhirschfeld/qmk_firmware/commit/c999cb2e2966da24c0f12ba9d23f3ea696aba661

**Comment by nwhirschfeld** [comment]
@zvecr Thanks a lot for your review!

File: keyboards/handwired/technicpad/info.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
+{
+    "manufacturer": "nwhirschfeld",
+    "keyboard_name": "TechnicPad",
+    "maintainer": "nwhirschfeld",
+    "bootloader": "caterina",
+    "bootmagic": {
+        "matrix": [0, 0]
+    },
```
Comment (zvecr) [inline keyboards/handwired/technicpad/info.json:8]: Duplicating defaults should be avoided
```suggestion
```

File: keyboards/handwired/technicpad/info.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
+{
+    "manufacturer": "nwhirschfeld",
+    "keyboard_name": "TechnicPad",
+    "maintainer": "nwhirschfeld",
+    "bootloader": "caterina",
```
Comment (zvecr) [inline keyboards/handwired/technicpad/info.json:5]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/handwired/technicpad/info.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    "matrix_pins": {
+        "direct": [
+            ["D1"],
+            ["D0"],
+            ["D4"],
+            ["C6"]
+        ]
+    },
+    "processor": "atmega32u4",
```
Comment (zvecr) [inline keyboards/handwired/technicpad/info.json:25]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/technicpad/info.json
Comment (fauxpark) [inline keyboards/handwired/technicpad/info.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/technicpad/info.json
Comment (nwhirschfeld) [reply keyboards/handwired/technicpad/info.json:1]: Thanks!
Resolved that in https://github.com/nwhirschfeld/qmk_firmware/commit/048e590f85f73b6418630753972fba4cf1c31532

**Review by fauxpark — APPROVED** [review]

File: keyboards/handwired/technicpad/info.json (hunk 7 above)
Comment (nwhirschfeld) [reply keyboards/handwired/technicpad/info.json]: solved in https://github.com/qmk/qmk_firmware/pull/23238/commits/808b271c2898a5b048c03d578a69c0c1fb882b6e

File: keyboards/handwired/technicpad/info.json (hunk 9 above)
Comment (nwhirschfeld) [reply keyboards/handwired/technicpad/info.json]: solved in https://github.com/qmk/qmk_firmware/pull/23238/commits/02352065d1f847a4a689679db243adc3d4d59e00

File: keyboards/handwired/technicpad/info.json (hunk 8 above)
Comment (nwhirschfeld) [reply keyboards/handwired/technicpad/info.json]: solved in https://github.com/qmk/qmk_firmware/pull/23238/commits/557f38498da3b8484ad4a540b8283b63eaa7fbaf

**Review by zvecr — APPROVED** [review]

**Pull request merged**